src/aggregate/aggregate_request.c

```diff
@@ -506,9 +502,8 @@ static int parseQueryArgs(ArgsCursor *ac, AREQ *req, RSSearchOptions *searchOpts
       ReturnedField *f = FieldList_GetCreateField(&req->outFields, name, path);
       f->explicitReturn = 1;
     }
+    FieldList_RestrictReturn(&req->outFields);
   }
-  // TODO: move inside the if?
-  FieldList_RestrictReturn(&req->outFields);

```

Comment:
are we sure that `HIGHLIGHT` and `SUMMARIZE` can't add non-explicit fields?

Reply:
I didn't understand the question

Reply:
This function actually did something only if the explicit flag was turned on, and I see that it is turned on inside the condition block. But is it possible for another function (like the highlight and summarize init functions) to turn this flag as well?

Reply:
The only place where we turn this flag on is inside this if